core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -437,12 +433,7 @@ private void handleKeycode(int keycode, int source) {
                 break;
             case KeyEvent.KEYCODE_MEDIA_PAUSE:
                 if (status == PlayerStatus.PLAYING) {
-                    mediaPlayer.pause(false, true);
-                }
-                if (UserPreferences.isPersistNotify()) {
-                    mediaPlayer.pause(false, true);
-                } else {
-                    mediaPlayer.pause(true, true);
+                    mediaPlayer.pause(() -> !UserPreferences.isPersistNotify(), false);
```

Comment:
I believe I understand now. First, the fact that it had two calls was simply a mistake, as far as I know (based on the commit that introduced it).
Then, the reinit only applies when streaming and using an intent (media buttons, notification, etc), so it means the AP UI isn't open and, running on background, we probably want to save some bandwidth.
It's fixed now.